config/config.go

```diff
@@ -21,6 +21,7 @@ type Config struct {
 	ObjFreshThreshold time.Duration // Ages below this uses theme.data.durationfresh coloring
 	Preset            Preset        // Color theme preset
 	Theme             Theme
+	Pager             string
```

Comment:
Currently you can't enable the pager via the config.

Maybe that should be possible?

How would that work? (Brainstorming a little)

What if just by setting the `pager` config (or `KUBECOLOR_PAGER` env var), then you don't need to set `--pager` flag?

Adding the flag `--pager=false` should then disable the pager.

Or is this simply overloading the config field? Maybe it should be a separate config field for enabling the pager, such as `pagerEnabled: true / false`. Gets quite verbose though, especially with the env var which would then be `KUBECOLOR_PAGERENABLED`...

Wdyt?

Reply:
I have now opted to use a `Paging` struct with `Enabled` and `Cmd`. So we have a config file like this:
```yaml
paging:
  enabled: true
  cmd: less -RF --mouse
```
And we have the following environment variables:
```
KUBECOLOR_PAGING_ENABLED
KUBECOLOR_PAGING_CMD
KUBECOLOR_PAGER
PAGER
```

I would have preferred "pager" over "paging" but then we can't rebind `KUBECOLOR_PAGER` anymore with Viper.

Reply:
Not really liking that.

Looking for inspiration in other tools, such at Git:

- setting config `core.pager less` enables pagination
- setting config `core.pager ""` (empty string) disables the pager
- can be overridden with flags `--paginate` and `--no-pager`
- also reads env var `GIT_PAGER` and falls back on `PAGER`

Or for example [bat](https://github.com/sharkdp/bat?tab=readme-ov-file#using-a-different-pager):

- control when to paginate with flags `--paging=always` or `--paging=never`
- control pager command with `--pager=less`
- reads env var `BAT_PAGER` or `PAGER`

I think it would be excessive to add both `KUBECOLOR_PAGER=less/more` and `KUBECOLOR_PAGING=true/false`.

For now I think we could settle with only using `KUBECOLOR_PAGER`, and rely on the var being set or not. And then have `--pager=less` flag override the env var, and a utility `--no-pager` flag to just disable it could be nice

Reply:
No worries about the back and forth! And it seemed I missed that the `--pager` flag was not working anymore (due to a typo). I'll adopt your suggestions.

I'm not aware of any plugins with paging flags. If it does happen in the future I think we would need to overwrite them anyway (we don't want kubectl to run its own pager).

Reply:
Kubectl should not run its own pager, but a kubectl plugin might want to.

Parsing flags and command at the same time could get tricky. Such as if you had the command:

```bash
kubectl myplugin --pager=1234
```

Right now kubecolor would just swallow that `--pager` flag, even though `kubectl-myplugin` might've had a flag called `--pager`.

If kubecolor's parsing was a bit smarter, it might've seen "myplugin", lookup `kubectl-myplugin`, realize that it's a plugin and therefore stop parsing its own flags.

Might be worth a thought. But perhaps we should just burn that bridge when we get to it. Users have an easy workaround anyway by just doing:

```bash
\kubectl myplugin --pager=1234
```

Reply:
@applejag The point I was trying to make is that if a plugin runs a pager (the most likely result of having such a flag), we have worse problems than the swallowing of the flag. (I guess in most cases the pager just won't run as it detects that stdout is not a terminal.) So indeed, let's wait until an actual problem arises.

Reply:
Yeah but if kubecolor detects that the subcommand is a plugin, then it basically turns itself into a noop and just passes that command the raw stdin, stdout, and stderr.

Reply:
Ah I didn't know kubecolor worked liked that.

Thinking about this, I think not swallowing flags is also bad. If you pass on a flag to kubectl that it doesn't recognize it will fail, so you'd have to pass and swallow in exactly the right situations. Perhaps we shouldn't be using flags at all?

Reply:
Using flags is OK. We just need to be careful with which ones we claim.

Adding `--kubecolor-` prefix would make it quite clunky. Like:

```bash
kubectl get pods --kubecolor-paging=auto
```

But at least it'd be way less receptacle to collisions.

Reply:
You know what, for the rare case `--pager`, `--paging`, or `--no-paging` collides, we could just add exceptions.

For now, let's start with a release candidate, and see how people like it.